src/node_sqlite.cc

```diff
@@ -506,6 +508,132 @@ void StatementSync::All(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(Array::New(isolate, rows.data(), rows.size()));
 }
 
+struct IterateCaptureContext {
+  int num_cols;
+  StatementSync* stmt;
+};
+
+void StatementSync::IterateReturnCallback(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  auto isolate = env->isolate();
+
+  Local<External> data = Local<External>::Cast(args.Data());
+  IterateCaptureContext* capture_context =
+      static_cast<IterateCaptureContext*>(data->Value());
+  auto stmt = capture_context->stmt;
+  sqlite3_reset(stmt->statement_);
+
+  LocalVector<Name> keys(isolate, {env->done_string(), env->value_string()});
+  LocalVector<Value> values(isolate,
+                            {Boolean::New(isolate, true), Null(isolate)});
+
+  Local<Object> result =
+      Object::New(isolate, Null(isolate), keys.data(), values.data(), 2);
+  args.GetReturnValue().Set(result);
+}
+
+void StatementSync::IterateNextCallback(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  auto isolate = env->isolate();
+
+  Local<External> data = Local<External>::Cast(args.Data());
+  IterateCaptureContext* capture_context =
+      static_cast<IterateCaptureContext*>(data->Value());
+  auto stmt = capture_context->stmt;
+  auto num_cols = capture_context->num_cols;
+
+  int r = sqlite3_step(stmt->statement_);
+  if (r != SQLITE_ROW) {
+    CHECK_ERROR_OR_THROW(
+        env->isolate(), stmt->db_->Connection(), r, SQLITE_DONE, void());
+    sqlite3_reset(stmt->statement_);
+
+    LocalVector<Name> keys(isolate, {env->done_string(), env->value_string()});
+    LocalVector<Value> values(isolate,
+                              {Boolean::New(isolate, true), Null(isolate)});
+
+    Local<Object> result =
+        Object::New(isolate, Null(isolate), keys.data(), values.data(), 2);
+    args.GetReturnValue().Set(result);
+    return;
+  }
+
+  LocalVector<Name> row_keys(isolate);
+  row_keys.reserve(num_cols);
```

Comment:
```suggestion
  LocalVector<Name> row_keys(isolate, num_cols);
```
?

Reply:
Here and below.

Reply:
for some reason, if I do that, it segfault so I'll keep `.reserve(num_cols)`

Reply:
I made the same mistake in my PR. The overload with a length is not equivalent to calling `.reserve()`. It initializes the values (with empty `Local`s in this case).